actors/user/src/main/java/org/sunbird/user/actors/UserProfileActor.java

```diff
@@ -0,0 +1,265 @@
+      privateDataMap = handlePrivateVisibility(privateList, esResult, esProfileVisibility);
+    }
+    if (privateDataMap != null && privateDataMap.size() >= esProfileVisibility.size()) {
+      // this will indicate some extra private data is added
+      UserUtility.updateProfileVisibilityFields(privateDataMap, esResult);
+    }
+    return esProfileVisibility;
+  }
+
+  private Map<String, Object> getProfileVisibilityByUserIdFromES(String userId) {
+    return ElasticSearchUtil.getDataByIdentifier(
+        ProjectUtil.EsIndex.sunbird.getIndexName(),
+        ProjectUtil.EsType.userprofilevisibility.getTypeName(),
+        userId);
+  }
+
+  private Map<String, Object> getUserByIdFromES(String userId) {
+    Map<String, Object> esResult =
+        ElasticSearchUtil.getDataByIdentifier(
+            ProjectUtil.EsIndex.sunbird.getIndexName(),
+            ProjectUtil.EsType.user.getTypeName(),
+            userId);
+    if (esResult == null || esResult.size() == 0) {
+      throw new ProjectCommonException(
+          ResponseCode.userNotFound.getErrorCode(),
+          ResponseCode.userNotFound.getErrorMessage(),
+          ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+    }
+    return esResult;
+  }
+
+  private void validateFields(List<String> values, String listType) {
+    // Remove duplicate entries from the list
+    // Visibility of permanent fields cannot be changed
+    if (CollectionUtils.isNotEmpty(values)) {
+      List<String> distValues = values.stream().distinct().collect(Collectors.toList());
+      Util.validateProfileVisibilityFields(distValues, listType, getSystemSettingActorRef());
+    }
+  }
+
+  /**
+   * This method will first removed the remove the saved private data for the user and then it will
+   * create new private data for that user.
+   *
+   * @param dataMap Map<String, Object> allData
+   * @param privateDataMap Map<String, Object> only private data.
+   * @param userId String
+   * @return boolean
+   */
+  private void updateDataInES(
+      Map<String, Object> dataMap, Map<String, Object> privateDataMap, String userId) {
+    ElasticSearchUtil.createData(
+        ProjectUtil.EsIndex.sunbird.getIndexName(),
+        ProjectUtil.EsType.userprofilevisibility.getTypeName(),
+        userId,
+        privateDataMap);
+    ElasticSearchUtil.createData(
+        ProjectUtil.EsIndex.sunbird.getIndexName(),
+        ProjectUtil.EsType.user.getTypeName(),
+        userId,
+        dataMap);
+  }
+
+  /**
+   * THis methods will update user private field under cassandra.
+   *
+   * @param userId Stirng
+   * @param privateFieldMap Map<String,String>
+   */
+  private void updateCassandraWithPrivateFiled(String userId, Map<String, String> privateFieldMap) {
+
+    User updateUserObj = new User();
+    updateUserObj.setId(userId);
+    updateUserObj.setProfileVisibility(privateFieldMap);
+    Response response = getUserDao().updateUser(updateUserObj);
```

Comment:
Move this method to UserDao.updateProfileVisibility.

Reply:
Not sure, this is already calling update User of UserDao. 
profilevisibility is a map within userobj itself, do we really want to expose different method to update one column??